litejpeg/core/rle/rlecore.py

```diff
@@ -0,0 +1,316 @@
+                                  self.dovalid.eq(0)))
+                    ).Else(
+                       # Else if a non-zero AC cofficient is detected than the
+                       # output is been generated with the amplitude equal to
+                       # that of the AC cofficient and the number of zeros are
+                       # been indicated as the Runlength.
+                       # Making the dvalid to be 1.
+                       accumulator.eq(sink.data),
+                       runlength.eq(zero_count),
+                       zero_count.eq(0),
+                       accumulator_temp.eq(accumulator + (-2*accumulator[11]*accumulator)),
+                       self.dovalid.eq(1)))
+        ]
+
+        self.sync += [
+            self.dovalid_next.eq(self.dovalid),
+            self.dovalid_next_next.eq(self.dovalid_next),
+        ]
+
+        self.sync += [
+
+            # Connecting the Datapath module to the main module.
+            self.source_inter.data[0:12].eq(accumulator),
+            self.source_inter.data[12:16].eq(runlength),
+            self.source_inter.data[16].eq(self.dovalid)
+
+        ]
+
+        self.sync += [
+            self.source.data.eq(self.source_inter.data)
+        ]
+
+
+class RunLength(PipelinedActor, Module):
+    """
+    This module will connect the RLE core datapath with the input
+    and output either from other modules or from the Test Benches.
+    The input is been taken from the sink and source and is been
+    transferred to the RLE core datapath by using read and write count.
+    The RLEdatapath will than calculate the number of zeros between two
+    consecutive non-zero numbers and give the output as runlength.
+
+    Attributes :
+    ------------
+    sink : 12 bits
+           receives data from the RLEmain containing the amplitude.
+    source : 17 bits
+             transmit output to the RLEmain
+             12 bits : amplitude
+             4 bits : runlength
+             1 bit : data_valid
+    write_swap, read_swap : 1 bit
+            To transmit the control from read to write or vice-versa in case
+            if one of them completes its execution, that is if all the data is
+            read or all the data is been written on the output.
+    """
+    def __init__(self):
+
+        # Connecting the module to the input and the output.
+        self.sink = sink = stream.Endpoint(
+                               EndpointDescription(block_layout(12)))
+        self.source = source = stream.Endpoint(
+                                   EndpointDescription(block_layout(17)))
+
+        # Adding PipelineActor to provide additional clock for the module.
+        PipelinedActor.__init__(self, datapath_latency)
+        self.latency = datapath_latency
+
+        # Connecting RLE submodule.
+        self.submodules.datapath = RLEDatapath()
+        self.comb += self.datapath.ce.eq(self.pipe_ce)
+
+        # Intialising the variables.
```

Comment:
You don't need to say "initialising the variables" it is clear from the code.

Reply:
DONE